src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,155 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> tail;
+    private Node<T> head;
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        linkBefore(value, findByIndex(index));
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T node : list) {
```

Comment:
better to rename as `value`, `node` is like inner implementation, not what the client pass as method parameter.